pyomo/contrib/doe/doe.py

```diff
@@ -791,16 +578,16 @@ def _finite_calculation(self, output_record, scena_gen):
         jac = {}
 
         # After collecting outputs from all scenarios, calculate sensitivity
-        for para in list(self.param.keys()):
+        for para in self.param.keys():
             # extract involved scenario No. for each parameter from scenario class
-            involved_s = scena_gen.scenario_para[para]
```

Comment:
Why did we switch from properties of `scena_gen` to a dictionary or dictionaries?